client-src/elements/utils.js

```diff
@@ -62,3 +62,42 @@ export function findFirstFeatureStage(intentStage, currentStage, fe) {
 export function flattenSections(stage) {
   return stage.sections.reduce((combined, section) => [...combined, ...section.fields], []);
 }
+
+/* Set up scrolling to a hash url (e.g. #id_explainer_links). */
+export function setupScrollToHash(pageElement) {
```

Comment:
To answer my own question and pose another one: since we don't know `container`, what if:
* The hash value was actually a dotted path like `id1.id2.id3` that could be split into parts to recursively drill into the layers of shadow DOM, or
* There were some well-known possible containers in the app, identified by 
    * ID paths of the containers, or 
    * by calling a `registerContainer`function like you have these setup calls, but with the narrower purpose of just adding a container element to a list of well-known containers without also defining a curried function.

Reply:
The scrollToElement function is bound internally to the pageElement, which is different for each page, so it can call the pageElement.shadowRoot.querySelector().  The one caller of this function that is stored on the global window object is the chromedash-app scrollIfHash function, so that caller doesn't know each of the pages it is on.   I didn't think we would want chromedash-app to import page elements from each page that it configures.

Reply:
What about my second comment?

Reply:
It seem like it should be possible to use the globally unique ids by searching in shadow DOM, but for closed shadow DOM, access is limited to the component that owns it.  So we would at least have to go through some api controlled by the compoents we are searching through.  But traversing a long list of components (the form fields) to find one with the id sounds time consuming and potentially more brittle (if we, for example, add a form field composed of two or more form eleents).  

We could more easily build a global map of ids to components, and just look up the component directly rather than doing any shadowRoot.querySelector lookups.

How about first we submit this PR to fix the immediate problems, and merge with Yanns change. Then I will be happy to do some refactoring.